### PR TITLE
leverage _set for small perf improvement in sitecaptain lookup

### PR DESCRIPTION
**Pull request by andy-d**

### CODE REVIEW

**Comment by andy-d** [comment]
@babybunny any thoughts on this?

File: room/staff.py [hunk 1]
```diff
@@ -25,20 +26,7 @@ class SelectProgram(webapp2.RequestHandler):
...
-      what_you_are_doing = "Select a Program to work on"
-      program_url_base = webapp2.uri_for('SelectProgram')
-      return common.Respond(self.request, 'select_program', locals())
-
-    if program not in common.PROGRAMS:
-      return http.HttpResponseError('program %s not in PROGRAMS' % program)
-    user.staff.program_selected = program
-    user.staff.put()
-    return webapp2.redirect_to('StaffHome')
+    webapp2.abort(404)
```
Comment (babybunny) [inline room/staff.py:29]: ?

File: room/staff.py [hunk 2]
```diff
@@ -188,15 +176,9 @@ def get(self):
...
-    # program.  use ancestor?
-    for sc in ndb_models.SiteCaptain.query():
-      sitecaptains_by_site.setdefault(sc.site.integer_id(), []).append(sc)
-    logging.info('loaded sitecaptains')
-    for s in entries:
-      k = s.key.integer_id()
-      if k in sitecaptains_by_site:
-        s.sitecaptains = sitecaptains_by_site[k]
+    for site in entries:
+      site.sitecaptains = list(site.sitecaptain_set)
```
Comment (babybunny) [inline room/staff.py:180]: There's a query under this sitecaptain_set property, hence the need to do all this dancing around in the first place. 
https://github.com/babybunny/rebuildingtogethercaptain/blob/93f90e72b723660d2f0f7d91fdf32b70aaae283d/room/ndb_models.py#L435

**Comment by andy-d** [comment]
Makes sense, just playing with things. Reading up more on NDB it looks like app engine now handles caching for you so assuming its up for awhile things should be warmed and I assume it's fast. I'll stop throwing spaghetti at the wall for speed boosts

**Pull request closed without merging**